### PR TITLE
Add deregistration_connection_termination to elb_target_group

### PR DESCRIPTION
**Pull request by marknet15**
##### SUMMARY
Adding support for the `deregistration_connection_termination` param in the `elb_target_group` module.

Along with this I've enabled and fixed up the integration tests.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
elb_target_group

##### ADDITIONAL INFORMATION

The API param is `deregistration_delay.connection_termination.enabled`

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/elbv2.html#ElasticLoadBalancingv2.Client.describe_target_group_attributes

### CODE REVIEW

File: plugins/modules/elb_target_group.py [hunk 1]
```diff
@@ -767,6 +780,9 @@ def create_or_update_target_group(connection, module):
     if deregistration_delay_timeout is not None:
         if str(deregistration_delay_timeout) != current_tg_attributes['deregistration_delay_timeout_seconds']:
             update_attributes.append({'Key': 'deregistration_delay.timeout_seconds', 'Value': str(deregistration_delay_timeout)})
+    if deregistration_connection_termination is not None:
+        if deregistration_connection_termination and current_tg_attributes.get('deregistration_delay_connection_termination_enabled') != "true":
```
Comment (marknet15) [inline plugins/modules/elb_target_group.py:784]: The AWS API seems to be a little inconsistent and needed `""` around the `bool` value like done here:
https://github.com/ansible-collections/community.aws/pull/913/files#diff-2c1c6b6915c4678f611809c13fca7baf3b17b239efd502403f050f52d2f03eafR787

**Comment by marknet15** [comment]
recheck

**Comment by marknet15** [comment]
recheck

**Comment by marknet15** [comment]
@markuman @alinabuzachis would you possible be able to review this one ? I'd be really keen to get it included in `3.1`

File: tests/integration/targets/elb_target/tasks/ec2_target.yml [hunk 2]
```diff
@@ -108,25 +107,39 @@
...
       elb_target_group:
         name: "{{ tg_name }}-nlb"
         health_check_port: 80
         protocol: tcp
         port: 80
-        vpc_id: '{{ vpc.vpc.id }}'
+        vpc_id: "{{ vpc.vpc.id }}"
         state: present
         target_type: instance
+        deregistration_delay_timeout: 30
```
Comment (markuman) [inline tests/integration/targets/elb_target/tasks/ec2_target.yml:141]: maybe set to a non-default value, to see that is processed correctly.

File: plugins/modules/elb_target_group.py [hunk 3]
```diff
@@ -897,6 +914,9 @@ def main():
     connection = module.client('elbv2', retry_decorator=AWSRetry.jittered_backoff(retries=10))
 
     if module.params.get('state') == 'present':
+        if module.params.get('protocol') in ['http', 'https', 'HTTP', 'HTTPS'] and module.params.get('deregistration_connection_termination', None):
```
Comment (markuman) [inline plugins/modules/elb_target_group.py:917]:
```suggestion
        if module.params.get('protocol').upper() in ['HTTP', 'HTTPS'] and module.params.get('deregistration_connection_termination', None):
```

Comment (markuman) [reply plugins/modules/elb_target_group.py:917]: Basically I'm fine with both. But it's already used like this: https://github.com/ansible-collections/community.aws/blob/main/plugins/modules/elb_target_group.py#L526

**Comment by marknet15** [comment]
recheck

**Comment by markuman** [comment]
Ah yeah, that introduced a bug now

`AttributeError: 'NoneType' object has no attribute 'upper''

**Review by markuman — APPROVED** [review]

**Review by alinabuzachis — APPROVED** [review]

**Pull request merged**